aten/src/ATen/core/jit_type.h

```diff
@@ -614,19 +542,32 @@ struct CAFFE2_API ProfiledTensorType : public TensorType {
     return cloned;
   }
 
-  ProfiledTensorTypePtr merge(ProfiledTensorTypePtr other) {
+  ProfiledTensorTypePtr merge(ProfiledTensorTypePtr other) const {
     auto scalar_type = merge_primitive(scalarType(), other->scalarType());
     auto dev = merge_primitive(device(), other->device());
     auto sz = sizes().merge(other->sizes());
     auto srs = strides().merge(other->strides());
     auto gr = merge_primitive(requiresGrad(), other->requiresGrad());
-    return ProfiledTensorType::create(scalar_type, dev, sz, srs, gr);
+    auto zero = merge_primitive(autogradZero(), other->autogradZero());
```

Comment:
does switching from an undefined to defined tensor make sense? Maybe, it should be an assert?
`TORCH_INTERNAL_ASSERT(autogradZero() == other->autogradZero())`
I actually went the same path and added the same field in my branch, but I wasn't sure if it makes sense for a tensor to switch.

Reply:
In this case it is merging a tensor which is known to be zero with one that that may/may not be zero. So the result is a tensor which may/may not be zero, which is what this implements.